src/gp_instance.cpp

```diff
@@ -174,3 +175,63 @@ void GPInstance::EstimateSBNParameters() {
   double marginal_log_lik = GetEngine()->GetLogMarginalLikelihood();
   std::cout << std::setprecision(9) << marginal_log_lik << std::endl;
 }
+
+RootedTreeCollection GPInstance::GetRootedTreeCollection() {
+
+  Tree::TreeVector tree_vector;
+  Node::NodePtrVec trees = dag_.GenerateAllTrees();
+  EigenVectorXd bl = engine_->GetBranchLengths();
+
+  // Leaves encoding to parent subsplit encoding.
+  std::unordered_map<const Node*, Bitset> leaves2subsplit_indexer;
+  for (size_t i = 0; i < trees.size(); i++) {
+    trees.at(i)->PreOrder([this, &leaves2subsplit_indexer](const Node *node)
+    {
+      GPDAGNode *dag_node = dag_.GetDagNode(node->Id());
+      if (!leaves2subsplit_indexer.count(node)) {
+        SafeInsert(leaves2subsplit_indexer, node, dag_node->GetBitset());
+      }
+
+      for (size_t i = 0; i < node->Children().size(); i++) {
+        size_t child_id = node->Children().at(i)->Id();
+        GPDAGNode *child_node = dag_.GetDagNode(child_id);
+        size_t i0 = dag_.GetGPCSPIndex(dag_node->GetBitset(false) + child_node->GetBitset());
```

Comment:
What's interesting about this is that you've added a default for GetGPCSPIndex, but then as far as I can tell the main purpose of the default is to raise an error if it's missing? Why not just raise an error if it's out of bounds in the GetGPCSPIndex function?

Reply:
If I'm not mistaken, the Node class sorts the children? So even though when I'm building the tree, I put the rotated child as the left child and the sorted child as the right child, this order is not held up later when we look up the index for PCSP.  I look up parent rotated subsplit + child subsplit and parent sorted subsplit + child subsplit and if I get the default for both, I raise an error as I expect exactly one to return a valid index.

Reply:
Got it. Let's rename to `GetPCSPIndexWithDefault` and move on. Thanks!

Reply:
Done.